server/api/utils/clients/iguazio.py

```diff
@@ -956,6 +961,17 @@ def _is_job_terminated(self, session: str, job_id: str) -> bool:
                 )
             raise
 
+    @staticmethod
+    def _resolve_final_error_message(error_message, result_message):
+        if error_message and result_message:  # If both are not empty
+            return f"{error_message}: {result_message}"
+        if not result_message:  # If result_message is empty
+            return error_message
+        if not error_message:  # If error_message is empty
+            return result_message
+        # If both are empty
+        return ""
```

Comment:
More concise, 
```suggestion
        result_message = result_message or ""
        error_message = error_message or ""
        return result_message or error_message
```

Reply:
This is not right because if `result_message` is empty and `error_message` is **not** empty it will return empty message instead of the `error_message` value